### DISCUSSION TITLE
Monorepo setup

### DISCUSSION THREAD

**Discussion by GuilleDF** [apache/devlake#7625] [Q&A]
Hi all! I'm trying to set up devlake to report DORA metrics for projects that are in a monorepo, but I'm not sure if this is possible with devlake. Ideally, I'd like to report individual metrics for each package in the monorepo, since they are essentially different projects. Here's my current setup:
Repo structure:
```
packages/
  - api/
  - sdk/
  - ... and others
```
In devlake, I created 2 projects:
```
- payments-api
- payments-sdk
```

For the sources, I'm using GitHub for source code management and for code review, and GitLab for CI/CD (we mirror our repos to GitLab to use Gitlab CI)

In the scope config, I'm able to configure the deployment rules so that for the 1st repo, only pipelines for tags like `api@.*` are considered deployments, and for the 2nd, tags like `sdk@.*`.

But, for metrics like [Median Lead Time for Changes](https://devlake.apache.org/docs/Metrics/LeadTimeForChanges) and similar, I imagine it will take into account every single PR made to the repo on both projects.

So even if we consistently merge, for example 1hr after each change, if in a month we make a lot of changes to the sdk, but very little to the api, the metrics for the api project will say the lead time for changes for the api is really high since it will consider changes to the sdk as "unreleased changes to the api".

Is there a way to filter PRs by the files they touch, so that they're only assigned to a project if they modify a specific set of files?

If not, I would also settle for any other convention we could adopt, like labelling PRs or passing the component name in PR titles (`feat(api): some feature`), etc.

**Comment by klesh** [comment]
For now, there is no way to filter PRs by the files they touch.
No other conventions are available at this point AFAIK, and the mono repo pattern is not supported officially.
